client/components/data/query-products-list/index.jsx

```diff
@@ -16,13 +16,13 @@ const request =
 		dispatch( requestProductsList( props ) );
 	};
 
-export function useQueryProductsList( { type = 'all', currency, persist } = {} ) {
+export function useQueryProductsList( { type = 'all', currency, persist, productSlugList } = {} ) {
 	const dispatch = useDispatch();
 
 	// Only runs on mount.
 	useEffect( () => {
-		dispatch( request( { type, currency, persist } ) );
-	}, [ dispatch, type, persist, currency ] );
+		dispatch( request( { type, currency, persist, productSlugList } ) );
```

Comment:
On `/start/do-it-for-me/new-or-existing-site`, the request is made with incorrect params:
<img width="286" alt="image" src="https://github.com/Automattic/wp-calypso/assets/5436027/34949ffc-99f7-4e1b-97f3-652cc27c3177">


Since `requestProductList` now accepts `query.product_slugs`, we should change this to:
```
dispatch( request( { type, currency, persist, product_slugs: productSlugList } ) );
```

Reply:
Nice catch.